usr/sbin/p11sak/p11sak.c

```diff
@@ -6254,6 +6607,90 @@ static int p11sak_pem_password_cb(char *buf, int size, int rwflag,
     return len;
 }
 
+/*
+ * Reads an openssl X509 from PEM-type input file. No error msgs here, 
+ * calling routine will issue any errors.
+ */
+static CK_RV p11sak_x509_from_pem_file(X509 **x509)
+{
+    X509 *x = NULL;
+    BIO *bio = NULL;
+    CK_RV rc;
+
+    bio = BIO_new_file(opt_file, "r");
+    if (bio == NULL) {
+        rc = CKR_FUNCTION_FAILED;
+        goto done;
+    }
+
+    x = PEM_read_bio_X509(bio, NULL, p11sak_pem_password_cb, NULL);
+    if (x == NULL) {
+        rc = CKR_FUNCTION_FAILED;
+        goto done;
+    }
+
+    *x509 = x;
+    rc = CKR_OK;
+
+done:
+    if (bio != NULL)
+        BIO_free(bio);
+
+    return rc;
+}
+
+/*
+ * Reads an openssl X509 from binary input file. No error msgs here, 
+ * calling routine will issue any errors.
+ */
+static CK_RV p11sak_x509_from_der_file(X509 **x509)
+{
+    CK_BYTE *value;
+    const CK_BYTE *tmp_value;
+    CK_ULONG value_len;
+    X509 *x;
+    struct stat sb;
+    FILE *fp = NULL;
+    CK_RV rc;
+
+    if (stat(opt_file, &sb) != 0)
+        return CKR_FUNCTION_FAILED;
```

Comment:
warnx here with printing the textual errno string (strerror())?
I guess the file must exist n any case, even if its not a DER file, and you retry to read the PEM file.

Reply:
Done.